### PR TITLE
Added support for using proxy servers

### PR DESCRIPTION
**Pull request by javiereguiluz**
I've tested it successfully with some free public proxy servers, but we should definitely test it with "real" proxy servers.

### CODE REVIEW

File: src/Symfony/Installer/DownloadCommand.php [hunk 1]
```diff
@@ -150,6 +151,25 @@ protected function download()
     }
 
     /**
+     * Returns the Guzzle client configured according to the system environment
+     * (e.g. it takes into account whether it should use a proxy server or not).
+     *
+     * @return Client
+     */
+    protected function getGuzzleClient()
```
Comment (xabbuh) [inline src/Symfony/Installer/DownloadCommand.php]: Is there a reason not to make this private?

**Comment by xabbuh** [comment]
The `NewCommand` class also uses the Guzzle client to download the list of available versions.

File: src/Symfony/Installer/DownloadCommand.php (hunk 1 above)
Comment (javiereguiluz) [inline src/Symfony/Installer/DownloadCommand.php]: Nice catch! Fixed. Thanks.

**Comment by javiereguiluz** [comment]
Nice catch! Fixed. Thanks.

File: src/Symfony/Installer/DownloadCommand.php (hunk 1 above)
Comment (sstok) [inline src/Symfony/Installer/DownloadCommand.php]: You can't call private methods in subclasses :)

Comment (xabbuh) [inline src/Symfony/Installer/DownloadCommand.php]: It wasn't used in another class the time I wrote the comment. :D

Comment (javiereguiluz) [inline src/Symfony/Installer/DownloadCommand.php]: You both are right! Thanks for reviewing the code.

**Comment by javiereguiluz** [comment]
I'm going to merge this now and release a new installer version. The code looks "safe enough" to not break anything if you are not using a proxy. And it also looks like we're going to solve the problem at least for some proxy users.

**Pull request closed without merging**